src/EditorFeatures/TestUtilities/Classification/AbstractClassifierTests.cs

```diff
@@ -40,12 +39,31 @@ public abstract class AbstractClassifierTests
            string code,
            string allCode,
            TestHost testHost,
-           ParseOptions parseOptions,
+           ParseOptions? parseOptions,
            params FormattedClassification[] expected)
         {
-            var start = allCode.IndexOf(code, StringComparison.Ordinal);
-            var length = code.Length;
-            var span = new TextSpan(start, length);
+            TextSpan span;
+            if (code != allCode)
+            {
+                var start = allCode.IndexOf(code, StringComparison.Ordinal);
+                var length = code.Length;
+                span = new TextSpan(start, length);
+            }
+            else
+            {
+                MarkupTestFile.GetSpans(allCode, out var rewrittenCode, out ImmutableArray<TextSpan> spans);
```

Comment:
new support so you can classify a portion of a file without having to specify the classifications for everything.  desirable as several tests add stubs for the StringSyntaxAttribute, and we don't want to have to write out all the classifications for that.